unstable/wlr-export-dmabuf-unstable-v1.xml

```diff
@@ -0,0 +1,233 @@
+           summary="indicates special frame features"/>
+      <arg name="format" type="uint"
+           summary="format of the layer (DRM_FORMAT_*)"/>
+      <arg name="mod_high" type="uint"
+           summary="drm format modifier, high"/>
+      <arg name="mod_low" type="uint"
+           summary="drm format modifier, low"/>
+      <arg name="num_objects" type="uint"
+           summary="indicates how many objects (FDs) the frame has (max 4)"/>
+      <arg name="num_planes" type="uint"
+           summary="indicates how many planes the frame has (max 4)"/>
+    </event>
+    <event name="object">
+      <description summary="object receiving callback">
+        Callback which serves to supply the client with the file descriptors
+        containing the data for each object.
+      </description>
+      <arg name="index" type="uint"
+           summary="index of the current object"/>
+      <arg name="fd" type="fd"
+           summary="fd of the current object"/>
+      <arg name="size" type="uint"
+           summary="size in bytes for the current object"/>
+    </event>
+    <event name="plane">
+      <description summary="plane info receiving callback">
+        Callback which supplies the client with plane information for each
+        layer.
+      </description>
+      <arg name="index" type="uint"
+           summary="index of the current plane"/>
+      <arg name="layer_index" type="uint"
+           summary="index of the current layer"/>
+      <arg name="object_index" type="uint"
+           summary="index of the object the plane applies to"/>
+      <arg name="offset" type="uint"
+           summary="starting point for the data in the object's fd"/>
+      <arg name="stride" type="uint"
+           summary="line size in bytes"/>
+    </event>
+    <event name="ready">
+      <description summary="indicates frame is available for reading">
+        Called as soon as the frame is presented, indicating it is available
+        for reading.
+        The timestamp is expressed as tv_sec_hi, tv_sec_lo, tv_nsec triples,
+        each component being an unsigned 32-bit value. Whole seconds are in
+        tv_sec which is a 64-bit value combined from tv_sec_hi and tv_sec_lo,
+        and the additional fractional part in tv_nsec as nanoseconds. Hence,
+        for valid timestamps tv_nsec must be in [0, 999999999].
+        The seconds part may have an arbitrary offset at start.
+      </description>
+      <arg name="tv_sec_hi" type="uint"
+           summary="high 32 bits of the seconds part of the timestamp"/>
+      <arg name="tv_sec_lo" type="uint"
+           summary="low 32 bits of the seconds part of the timestamp"/>
+      <arg name="tv_nsec" type="uint"
+           summary="nanoseconds part of the timestamp"/>
+    </event>
+
+    <enum name="abort_reason">
+      <description summary="abort reason">
+        Indicates reason for aborting the frame.
+      </description>
+      <entry name="invalid_source" value="0" since="1"
```

Comment:
I need a reason to say "the DMA-BUF export failed, I'm not sure why, but you may try again and maybe it'll work".

Maybe we could have a boolean saying if the failure is temporary instead?

Reply:
Let me reformulate my question: do we need the receiver to know the reason of the failure rather than to know if he needs to retry or not? Does this enum add any value to the protocol?